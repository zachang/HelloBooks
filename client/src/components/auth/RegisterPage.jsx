import React from 'react';
import { Link } from 'react-router';
import HomepageHeader from '../common/HomepageHeader.jsx';
import RegForm from './RegForm.jsx';

/**
 * RegisterPage class declaration
 *
 * @class RegisterPage
 *
 * @extends {React.Component}
 */
export default class RegisterPage extends React.Component {
  /**
   * class constructor
   *
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.state = {
      showLoginText: true
    };
  }

  /**
   * @method componentDidUpdate
   *
   * @return {void} void
   */
  componentDidMount() {
    $('.dropdown-button').dropdown();
  }

  /**
   * Renders RegisterPage component
   *
   * @return {XML} JSX
   */
  render() {
    return (
      <div className='row main-row main-div'>
        <HomepageHeader btnText={this.state.showLoginText}/>

        <div className='section no-pad-bot' id='index-banner'>
          <div className='container textContent'>
            <div className='row'>
              <div className='col s8 offset-s2 regsmContent'>
                <RegForm/>
                <h6 className='infoText'>
                  Already have an account...?
                  <Link to='/login'>
                    Login
                  </Link>
                </h6>
                <br/><br/>
              </div>
            </div>
          </div>

        </div>
      </div>
    );
  }
}