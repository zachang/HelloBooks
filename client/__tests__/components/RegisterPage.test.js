import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import RegisterPage from '../../src/components/auth/RegisterPage.jsx';

describe('<RegisterPage/>', () => {
  const props = {
    showLoginText: false
  };

  const shallowWrapper = shallow(<RegisterPage {...props}/>);
  const tree = toJson(shallowWrapper);

  it('renders <RegisterPage /> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it(`should check if the 'HomepageHeader' is defined`, () => {
    expect(shallowWrapper.find('HomepageHeader').length).toBe(1);
  });

  it(`should check if the 'RegForm' is defined`, () => {
    expect(shallowWrapper.find('Connect(RegForm)').length).toBe(1);
  });
});

