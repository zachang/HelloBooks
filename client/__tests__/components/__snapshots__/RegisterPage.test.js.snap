// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage/> renders <RegisterPage /> component 1`] = `
<div
  className="row main-row main-div"
>
  <HomepageHeader
    btnText={true}
  />
  <div
    className="section no-pad-bot"
    id="index-banner"
  >
    <div
      className="container textContent"
    >
      <div
        className="row"
      >
        <div
          className="col s8 offset-s2 regsmContent"
        >
          <Connect(RegForm) />
          <h6
            className="infoText"
          >
            Already have an account...?
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/login"
            >
              Login
            </Link>
          </h6>
          <br />
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
